lib/openhab/core/configuration.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "forwardable"
+require_relative "emulate_hash"
+
+module OpenHAB
+  module Core
+    java_import org.openhab.core.config.core.Configuration
+
+    #
+    # {Configuration} represents openHAB's {org.openhab.core.config.core.Configuration} data
+    # with the full interface of {::Hash}.
+    #
+    # All keys are converted to strings.
+    #
+    class Configuration
+      include EmulateHash
+      extend Forwardable
+
+      def_delegator :get_properties, :to_hash
+      def_delegator :to_h, :inspect
+
+      # @!visibility private
+      alias_method :to_s, :inspect
+
+      # @!visibility private
+      def fetch(key, *default_value, &block)
+        if (result = get(key.to_s))
+          result
+        elsif block
+          yield key
+        elsif default_value.length > 1
+          raise ArgumentError, "Wrong number of arguments (given #{default_value.length + 1}, expected 1..2)"
+        elsif !default_value.empty?
+          default_value.first
+        else
+          raise KeyError, "Key not found #{key.inspect}"
+        end
+      end
+
+      # @!visibility private
+      def store(key, value)
+        put(key.to_s, value)
+      end
+
+      # @!visibility private
+      def delete(key)
+        remove(key.to_s)
+      end
+
+      # @!visibility private
+      def key?(key)
+        contains_key(key.to_s)
+      end
+      alias_method :has_key?, :key?
+      alias_method :include?, :key?
+
+      # @!visibility private
+      def replace(new_pairs)
+        set_properties(new_pairs.to_h.transform_keys(&:to_s))
+        self
+      end
+
+      # @!visibility private
+      alias_method :keys, :key_set
+
+      # @!visibility private
+      def dup
+        self.class.new(self)
+      end
+
+      # @!visibility private
+      def ==(other)
+        return to_hash == other.transform_keys(&:to_s) if other.is_a?(Hash)
```

Comment:
`to_hash == other.to_hash if other.respond_to?(:to_hash)` (with ActiveSupport's indifferent access, comparisons don't convert keys:

```ruby
irb(main):001:0> i = { a: 1, "b" => 2 }.with_indifferent_access
=> {"a"=>1, "b"=>2}
irb(main):002:0> h = { a: 1, "b" => 2 }
=> {:a=>1, "b"=>2}
irb(main):003:0> i == h
=> false
irb(main):004:0> h2 = { "a" => 1, "b" => 2 }
=> {"a"=>1, "b"=>2}
irb(main):005:0> i == h2
=> true
```
)

and since Configuration will respond to `to_hash`, but it creates a temporary, it should go _above_ this line (`return equals(other) if other.is_a?(self.class)`)